sync/storage/batch/batch.go

```diff
@@ -109,12 +124,17 @@ func (s *batchStorageSync) syncBucket(ctx context.Context, lastSuccessfulRun tim
 	if errCount > 0 {
 		s.logger.Error().Str("bucket", bucketID).Msgf("%d failure(s) out of %d file(s) to sync", errCount, syncCount)
 		errCh <- &syncError{bucketID, errors.Errorf("%d failure(s) out of %d file(s) to sync in bucket %s", errCount, syncCount, bucketID)}
+		<-rateLimit
 		return
 	}
+	<-rateLimit
+
 	errCh <- nil
 }
 
-func (s *batchStorageSync) syncFile(ctx context.Context, lastSuccessfulRun time.Time, bucketID, fileID string, errCh chan *syncError) {
+func (s *batchStorageSync) syncFile(ctx context.Context, lastSuccessfulRun time.Time, bucketID, fileID string, errCh chan *syncError, rateLimit chan bool) {
+	rateLimit <- true
```

Comment:
perhaps it would be safer to free the rateLimit slot with `defer` to make it more error prone

Reply:
Good idea :) done